src/components/GroceryItem/GroceryItem.js

```diff
@@ -1,9 +1,54 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { firestore } from '../../lib/firebase.js';
 
 const GroceryItem = ({ item }) => {
+  const groceryItem = firestore.collection('groceryItems').doc(item.id);
+  const [isChecked, setChecked] = useState(false);
+
+  const oneday = 60 * 60 * 24 * 1000;
+
+  useEffect(() => {
+    if (item.purchaseDate > Date.now() - oneday) {
+      setChecked(true);
+    }
+  }, []);
+
+  const setNewDate = () => {
+    return groceryItem
+      .update({
+        purchaseDate: Date.now(),
+      })
+      .then(() => {
```

Comment:
We are using async/await in other parts of the code. Not necessary but would consider using it here `try { await groceryItem.update... } catch (error) {/...`